azure-pipelines.yml

```diff
@@ -183,7 +183,8 @@ jobs:
     - task: DotNetCoreCLI@2
       displayName: Package Microsoft.Spark.Worker
       inputs:
-        command: msbuild
+        command: custom
+        custom: msbuild
         projects: eng\PackageWorker.proj
```

Comment:
Why does this not build in the previous step?

Reply:
I didn't fold packaging the worker into build/pack phase because I didn't see sign tool having support for `.tar.gz` files.  `PackageWorker.proj` builds 3 archives:

- 2 `.zip` files for Windows.
  - netfx
  - netcoreapp
- 1 `.tar.gz` file for Linux netcoreapp

Granted, it probably isn't critical that the managed assemblies in the `.tar.gz` file aren't signed. But it "feels wrong" to ship unsigned managed binaries, even if they are only going to be used on non-Windows.

Thoughts? Do you think we could ship unsigned binaries in the `.tar.gz` files? Or maybe we can add support in arcade for `.tar.gz` files?

As for keeping the packaging happening after signing, but doing it in the same build step, I guess I didn't know how to do that "the arcade way". How do I say:

`eng\common\build.ps1 -restore -sign --??` - what goes in `--??` to build an extra project?

Can I do:

`eng\common\build.ps1 -restore -sign -projects eng\PackageWorker.proj`

? Will that call signing first, and then build the `eng\PackageWorker.proj`? I thought it would be the other way around - build the project, then sign. Which is not what we want.

https://github.com/dotnet/arcade/blob/b5016f5688dc8ca9f3e4811ee7e2e86ad8907a40/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L215-L243

I guess, looking at that code now, I see the `AfterSigning.proj`. I could hook into that to do the `PackageWorker`. Thoughts?

Reply:
Could you file an issue on Arcade to enable signing artifacts in `.tar.gz`? I don't know if there is a good .NET API that allows us to re-package `.tar.gz` but we should look into it.

Reply:
Done - https://github.com/dotnet/arcade/issues/2915.